FBSnapshotTestCase/FBSnapshotTestCase.m

```diff
@@ -122,17 +123,52 @@ - (NSString *)snapshotVerifyViewOrLayer:(id)viewOrLayer
       }
     }
   }
-  
-  if (!testSuccess) {
-    return [NSString stringWithFormat:@"Snapshot comparison failed: %@", errors.firstObject];
-  }
+    
+  [self addAttachmentsWithErrors:errors identifier:identifier];
+    
   if (self.recordMode) {
-    return @"Test ran in record mode. Reference image is now saved. Disable record mode to perform an actual snapshot comparison!";
+    if (errors.count > 0) {
+      return [NSString stringWithFormat:@"Snapshot comparison failed: %@", errors.firstObject];
+    } else {
+      return @"Test ran in record mode. Reference image is now saved. Disable record mode to perform an actual snapshot comparison!";
+    }
+  } else if (!testSuccess) {
+    return [NSString stringWithFormat:@"Snapshot comparison failed: %@", errors.firstObject];
   }
 
   return nil;
 }
 
+- (void) addAttachmentsWithErrors:(NSArray<NSError *> *)errors identifier:(NSString *)identifier {
```

Comment:
An internal method should be prefixed with `_`, there should be no space between the return type and the start of the method name, and the first opening brace should be on a new line